assets/js/data/checkout/reducers.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * External dependencies
+ */
+import type { Reducer } from 'redux';
+
+import { PaymentResult } from '@woocommerce/types';
+/**
+ * Internal dependencies
+ */
+import { ACTION_TYPES as types } from './action-types';
+import { STATUS } from './constants';
+import { defaultState } from './default-state';
+import { CheckoutAction } from './actions';
+
+const reducer: Reducer = ( state = defaultState, action: CheckoutAction ) => {
+	let newState = state;
```

Comment:
It's because we need access to the previous state and incoming state in the [last if statement in the reducer](https://github.com/woocommerce/woocommerce-blocks/pull/6455/files#diff-d8eced2af1a16308787c61313fb6396eb4b5a53c6e64021685d364095dd0e640R176)